syntax = "proto3";

package ProcessMetricas;




message ProcessMetrics {
  // Campos básicos do processo
  int32 pid = 1;                          
  string name = 2;                        
  string user = 3;                        
  string status = 5;
  
  // Campos adicionais do processo                 
  int32 num_threads = 7;                  
  float cpu_percent = 8;                  
  int32 num_child_processes = 9;           
  float memory_percent = 10;
  
  // Campos de I/O e FDs
  uint64 read_bytes = 11;                
  uint64 write_bytes = 12;                
  int32 num_fds = 13;                     
  int32 nice = 14;

  // Campos de timestamp, boottime, timestartprocess
  string timestamp = 4;
  string boottime = 15;
  string timestartprocess = 16;
   
  // Campo para o IP do host                  
  string hostip= 17;
  
};

message ProcessMetricsList {
  repeated ProcessMetrics processes = 1;  
};

message KernelDistro {
  string kernel_version = 1;              
  string distro_name = 2;                 
}

// Mensagem que representa um programa instalado
message InstalledProgram {
  string name = 1;       // Nome do pacote/programa
  string version = 2;    // Versão do pacote/programa
}

// Mensagem que representa a lista de todos os programas
message InstalledProgramList {
  repeated InstalledProgram programs = 1;  // Lista de programas
}






